### PR TITLE
More work on the SVG view, prepping it for selecting and manipulating

### PR DESCRIPTION
**Pull request by jgao222**
Restructured some code but it will need to be restructured again later.

Changed the aspects of the camera to make it start at a distance that looks at the FOLD from a consistent distance away based on how big the 3D shape is.

SVG detects clicks on it, there are functions that handle this so can be used as scaffolding for building on selection behavior.
Added padding to SVG so clicks outside slightly still count.

### CODE REVIEW

File: external/fold.js [hunk 1]
```diff
@@ -381,6 +381,8 @@ convert.edges_vertices_to_edges_faces_edges = function(fold) {
     ];
   }
   ref1 = fold.faces_vertices;
+  // NOT PART OF ORIGINAL LIBRARY
+  fold.faces_edges = [];
```
Comment (sstadnicki) [inline external/fold.js:385]: What's the purpose of this line?

File: index.html [hunk 2]
```diff
@@ -6,46 +6,32 @@
...
     <body>
         <div class='parent flex-parent'>
             <div class='child flex-child'>
                 <canvas id="glCanvas" width="640" height="480"></canvas>
             </div>
             <div class='child flex-child'>
-                <svg id="svg" viewBox="0 0 100 100" width="800" height="600" preserveAspectRatio="none">
-
-
+                <svg id="svg" viewBox="0 0 800 600" width="800" height="600" preserveAspectRatio="none">
```
Comment (sstadnicki) [inline index.html]: I haven't double-checked how the SVG rendering is handled but you might want this to be square just to match the (square) sheet dimensions.

**Review by sstadnicki — APPROVED** [review]

File: external/fold.js (hunk 1 above)
Comment (jgao222) [reply external/fold.js:385]: There's a bug in the FOLD API where this method is supposed to create the `faces_edges` attribute if it doesn't exist, but never initializes it like it does in other methods, which leads it to try to assign values to something that doesn't exist. This line initializes that.

Comment (sstadnicki) [reply external/fold.js:385]: Ahhh, that makes sense. Thank you!

**Pull request merged**